src/MustardBlack.Assets.Nuglify/NuglifyJavascriptPreprocessor.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using MustardBlack.Assets.Javascript;
+using NUglify;
+
+namespace MustardBlack.Assets.Nuglify
+{
+    public class NuglifyJavascriptPreprocessor : IJavascriptPreprocessor
+    {
+	    public string Process(IEnumerable<AssetContent> assets)
+	    {
+		    // Check that the individual assets compress in on their own.
+		    foreach (var asset in assets)
+			    CompressJavascript(asset.Contents, asset.Path);
+		    
+		    var builder = new StringBuilder();
+		    foreach (var asset in assets)
+			    builder.AppendLine(asset.Contents);
+		    
+		    return CompressJavascript(builder.ToString(), null);
+	    }
+
+	    static string CompressJavascript(string javascriptString, string assetPath)
+	    {
+		    try
+		    {
+			    return Uglify.Js(javascriptString).Code;
```

Comment:
Looks like Nuglify returns errors rather than throws exceptions (although obvs better to have the catch anyway).
So think a check for errors needs adding
```suggestion
			    var nuggled = Uglify.Js(javascriptString);
			    if(!nuggled.HasErrors)
			            return nuggled.Code;
			    	    
	                    //parse and console out the errors here (you get lots of context in the error, see image)
```
![image](https://user-images.githubusercontent.com/1648478/86758411-c2ade080-c03b-11ea-888e-ff62219511f2.png)

Reply:
Sorted.